### PR TITLE
Small Makefile cleanup.

### PR DESCRIPTION
**Pull request by mpoullet**
- add target run_assistant to the phony targets
- put source files together in _SRCS groups
- add missing run_assistant_file.o to ASSISTANT_O target
- do not link with libcurl

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,22 +75,23 @@ CXXFLAGS += $(ALSA_CFLAGS)
...
+ASSISTANT_AUDIO_O = $(CORE_SRCS:.cc=.o) $(AUDIO_SRCS:.cc=.o) \
+                    ./src/audio_input_file.o ./src/run_assistant_audio.o
+ASSISTANT_FILE_O  = $(CORE_SRCS:.cc=.o) $(AUDIO_SRCS:.cc=.o) \
+                    ./src/audio_input_file.o ./src/run_assistant_file.o
+ASSISTANT_TEXT_O  = $(CORE_SRCS:.cc=.o) \
+                    ./src/run_assistant_text.o
 
 .PHONY: all
-all: run_assistant
+all: run_assistant_audio run_assistant_file run_assistant_text
```
Comment (Fleker) [inline Makefile]: I would run this with `make all`?

I am still keeping the `run_assistant` logic for [`tests/build.sh`](https://github.com/googlesamples/assistant-sdk-cpp/blob/master/tests/build.sh#L50) which I run on a CI server before submitting a change. That way I can make sure they all compile with a single command.

If we were to change this command, we'd need to update `build.sh` as well.

File: Makefile [hunk 2]
```diff
@@ -75,22 +75,23 @@ CXXFLAGS += $(ALSA_CFLAGS)
...
-	./src/run_assistant_audio.o ./src/run_assistant_text.o
-ASSISTANT_AUDIO_O = $(AUDIO_SRCS:.cc=.o) ./src/audio_input_file.o ./src/base64_encode.o ./src/json_util.o \
-	./src/run_assistant_audio.o
-ASSISTANT_FILE_O = $(AUDIO_SRCS:.cc=.o) ./src/audio_input_file.o ./src/base64_encode.o ./src/json_util.o \
-	./src/run_assistant_file.o
-ASSISTANT_TEXT_O = ./src/json_util.o ./src/base64_encode.o ./src/run_assistant_text.o
+CORE_SRCS = ./src/base64_encode.cc ./src/json_util.cc
+
+ASSISTANT_O       = $(CORE_SRCS:.cc=.o) $(AUDIO_SRCS:.cc=.o) \
+                    ./src/audio_input_file.o ./src/run_assistant_audio.o ./src/run_assistant_text.o
```
Comment (Fleker) [inline Makefile]: I think this is missing `./src/run_assistant_file.o`

File: Makefile (hunk 1 above)
Comment (mpoullet) [reply Makefile]: Understood. I'd add it to .PHONY tough.

File: Makefile (hunk 2 above)
Comment (mpoullet) [reply Makefile]: I've added it, but it was initially missing too.

File: Makefile [hunk 3]
```diff
@@ -75,20 +75,24 @@ CXXFLAGS += $(ALSA_CFLAGS)
...
-	./src/run_assistant_audio.o ./src/run_assistant_text.o
-ASSISTANT_AUDIO_O = $(AUDIO_SRCS:.cc=.o) ./src/audio_input_file.o ./src/base64_encode.o ./src/json_util.o \
-	./src/run_assistant_audio.o
-ASSISTANT_FILE_O = $(AUDIO_SRCS:.cc=.o) ./src/audio_input_file.o ./src/base64_encode.o ./src/json_util.o \
-	./src/run_assistant_file.o
-ASSISTANT_TEXT_O = ./src/json_util.o ./src/base64_encode.o ./src/run_assistant_text.o
+CORE_SRCS = ./src/base64_encode.cc ./src/json_util.cc
+
+ASSISTANT_O       = $(CORE_SRCS:.cc=.o) $(AUDIO_SRCS:.cc=.o) \
+                    ./src/audio_input_file.o ./src/run_assistant_audio.o ./src/run_assistant_file.o ./src/run_assistant_text.o
```
Comment (Fleker) [inline Makefile]: Should we put this into its own _SRCS?

Comment (mpoullet) [reply Makefile]: I've changed it so that each source code filename appears only once in the Makefile to ease maintainability.

File: Makefile [hunk 4]
```diff
@@ -61,11 +61,11 @@ CXXFLAGS += -std=c++11 $(GRPC_GRPCPP_CFLAGS)
 ifeq ($(SYSTEM),Darwin)
 LDFLAGS += $(GRPC_GRPCPP_LDLAGS) \
            -lgrpc++_reflection -lgrpc_cronet \
-           -lprotobuf -lpthread -ldl -lcurl
```
Comment (Fleker) [inline Makefile]: Is `lcurl` not needed?

Comment (mpoullet) [reply Makefile]: Well, I can definitely build and run the samples without cURL (on Ubuntu 16.04 and on diverse ARM-based boards). I don't know why it's there in the first place. I can't find any references to cURL API functions in the sources. cURL is no requirement for gRPC either. Putting `-lcurl` in the Makefile is misleading as it implies you need to have cURL available to use the SDK.

Comment (Fleker) [reply Makefile]: If it builds without `-lcurl` then that's great.

**Comment by mpoullet** [comment]
Very cool. Thanks a lot!

**Pull request merged**